NextGenSoftware.OASIS.API.Providers.EOSIOOASIS/EOSIOOASIS.cs

```diff
@@ -25,24 +32,163 @@ public List<IPlayer> GetPlayersNearMe()
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "email",
+                        new_value = profile.Email
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setstrval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "title",
+                        new_value = profile.Title
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setstrval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "firstname",
+                        new_value = profile.FirstName
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setstrval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "lastname",
+                        new_value = profile.LastName
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setstrval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "dob",
+                        new_value = profile.DOB
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setstrval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetStringValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "playeraddr",
+                        new_value = profile.PlayerAddress
+                    }
+                ));
+
+            actions.Add(new ActionUtility(chainApi.GetHost().AbsoluteUri).GetActionObject("setintval", profile.Username, "active", OASIS_EOSIO_ACCOUNT,
+                    new EOSIOSetIntValueParams()
+                    {
+                        eosio_acc = profile.Username,
+                        field_name = "karma",
+                        new_value = profile.Karma
+                    }
+                ));
```

Comment:
Why are the actions repeated again and again?

Reply:
That's because in usual it's rare to set all the properties of the account at once except when creating the acc. So I thought that having an action to set an individual field would be better. 
So when creating the account, I need to add several actions to the transaction to set all the fields one by one.

Reply:
Oh right, OK. Why can you not set them all at once? Would be a lot less code and be more efficient! ;-) Is this some sort of limitation with EOS?

Reply:
Of course, I can add an action to set all of those fields at once, but it's unnecessary as I can push many actions in a single transaction.

Reply:
Yes I would prefer they were all set at once... doesn't make sense to do it any other way... be less code then... thanks... get this done and then I can merge the changes in thanks...